tests/core/rkyv.rs

```diff
@@ -5,16 +5,16 @@ use na::{
     Rotation3, Similarity3, SimilarityMatrix2, SimilarityMatrix3, Translation2, Translation3,
 };
 use rand;
-use rkyv::{Deserialize, Infallible};
 
 macro_rules! test_rkyv_same_type(
     ($($test: ident, $ty: ident);* $(;)*) => {$(
         #[test]
         fn $test() {
             let value: $ty<f32> = rand::random();
-			let bytes = rkyv::to_bytes::<_, 256>(&value).unwrap();
+			let bytes = rkyv::to_bytes::<rkyv::rancor::Error>(&value).unwrap();
 
-			let archived = rkyv::check_archived_root::<$ty<f32>>(&bytes[..]).unwrap();
+            let archived = rkyv::access::<<$ty<f32> as rkyv::Archive>::Archived, rkyv::rancor::Error>(&bytes)
+        .unwrap();
             // Compare archived and non-archived
 			assert_eq!(archived, &value);
 
```

Comment:
The debug check underneath is adding a constraint on `Archived` to be `Debug`, which was not explicit before, I think that's what we'll want to do like what I did for `Matrix` through `archive_bounds(<S as Archive>::Archived: core::fmt::Debug)`, but it feels like it's adding potentially unnecessary constraints.

Could a manual implementation could implement Debug only for compatible types ?

Reply:
going through ~`as = Matrix<Archived::T>` and leveraging `Portable` is probably the best path forward, but I'm not sure how exactly.